src/core/Mixer.cpp

```diff
@@ -487,12 +438,70 @@ const surroundSampleFrame * Mixer::renderNextBuffer()
 
 	m_profiler.finishPeriod( processingSampleRate(), m_framesPerPeriod );
 
-	return m_readBuf;
+	return m_outputBufferRead;
 }
 
 
 
 
+void Mixer::swapBuffers()
+{
+	m_inputBufferWrite = (m_inputBufferWrite + 1) % 2;
+	m_inputBufferRead = (m_inputBufferRead + 1) % 2;
+	m_inputBufferFrames[m_inputBufferWrite] = 0;
+
+	std::swap(m_outputBufferRead, m_outputBufferWrite);
+	BufferManager::clear(m_outputBufferWrite, m_framesPerPeriod);
+}
+
+
+
+
+void Mixer::handleMetronome()
+{
+	static tick_t lastMetroTicks = -1;
+
+	Song * song = Engine::getSong();
+	Song::PlayModes currentPlayMode = song->playMode();
+
+	bool metronomeSupported = currentPlayMode == Song::Mode_PlayPattern ||
+					 currentPlayMode == Song::Mode_PlaySong ||
+					 currentPlayMode == Song::Mode_PlayBB;
```

Comment:
Don't align with tabs because they won't always appear the same to everyone. We should just make the whole expression indented with a single tab instead.
```suggestion
	bool metronomeSupported =
		currentPlayMode == Song::Mode_PlayPattern
		|| currentPlayMode == Song::Mode_PlaySong
		|| currentPlayMode == Song::Mode_PlayBB;
```